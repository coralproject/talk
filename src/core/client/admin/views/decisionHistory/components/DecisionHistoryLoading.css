.container {
  padding: var(--spacing-unit);
  min-height: 30px;
}
