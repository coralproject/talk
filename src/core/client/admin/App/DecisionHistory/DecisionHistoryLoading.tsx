import React, { FunctionComponent } from "react";

import { Delay, Flex, Spinner } from "coral-ui/components/v2";

import Main from "./Main";
import Title from "./Title";

import styles from "./DecisionHistoryLoading.css";

const DecisionHistoryLoading: FunctionComponent = () => (
  <div data-testid="decisionHistory-loading-container">
    <Title />
    <Main>
      <Flex
        justifyContent="center"
        alignItems="center"
        className={styles.container}
      >
        <Delay>
          <Spinner size="sm" />
        </Delay>
      </Flex>
    </Main>
  </div>
);

export default DecisionHistoryLoading;
