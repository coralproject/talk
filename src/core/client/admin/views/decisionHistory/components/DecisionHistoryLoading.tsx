import React, { StatelessComponent } from "react";

import { Delay, Flex, Spinner } from "talk-ui/components";

import styles from "./DecisionHistoryLoading.css";
import Main from "./Main";
import Title from "./Title";

const DecisionHistoryLoading: StatelessComponent = () => (
  <div data-test="decisionHistory-loading-container">
    <Title />
    <Main>
      <Flex
        justifyContent="center"
        alignItems="center"
        className={styles.container}
      >
        <Delay>
          <Spinner size="sm" />
        </Delay>
      </Flex>
    </Main>
  </div>
);

export default DecisionHistoryLoading;
