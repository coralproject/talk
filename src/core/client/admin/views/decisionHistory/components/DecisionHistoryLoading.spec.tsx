import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import DecisionHistoryLoading from "./DecisionHistoryLoading";

it("renders correctly", () => {
  const renderer = createRenderer();
  renderer.render(<DecisionHistoryLoading />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
