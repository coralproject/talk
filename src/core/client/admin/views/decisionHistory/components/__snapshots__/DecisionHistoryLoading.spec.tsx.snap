// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-test="decisionHistory-loading-container"
>
  <Title />
  <Main>
    <withPropsOnChange(Flex)
      alignItems="center"
      className="DecisionHistoryLoading-container"
      justifyContent="center"
    >
      <Delay
        ms={500}
      >
        <withPropsOnChange(Spinner)
          size="sm"
        />
      </Delay>
    </withPropsOnChange(Flex)>
  </Main>
</div>
`;
